apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestion.ts

```diff
@@ -136,6 +143,179 @@ router.post(
+          }
+
+          const question_prompt = data.questionHtml.split('<script>', 2)[0];
+
+          // Call OpenAI API
+          const completion = await openai.chat.completions.create({
+            messages: [
+              {
+                role: 'system',
+                content: `You are an instructor for a course, and you are grading assignments. You should always return the grade using a json object of 2 parameters: grade and feedback. The grade should be an integer between 0 and 100. 0 being the lowest and 100 being the highest, and the feedback should be why you give this grade, or how to improve the answer. You can say correct or leave blank when the grade is close to 100. `,
+              },
+              {
+                role: 'user',
+                content: `Question: \n${question_prompt} \nAnswer: \n${student_answer} \nHow would you grade this? Please return the json object.`,
+              },
+            ],
+            model: 'gpt-3.5-turbo',
+          });
+
+          let msg = '';
+          try {
+            if (completion.choices[0].message.content === null) {
+              error_count++;
+              continue;
+            }
+            const gpt_answer = JSON.parse(completion.choices[0].message.content);
+            const update_result = await manualGrading.updateInstanceQuestionScore(
+              res.locals.assessment.id,
+              instance_question.id,
+              submission.id,
+              req.body.modified_at,
+              {
+                score_perc: gpt_answer.grade,
+                feedback: { manual: gpt_answer.feedback },
+                // NEXT STEPS: rubrics
+              },
+              '1',
+            );
+            msg = `Bot grades for ${instance_question.id}: ${gpt_answer.grade}`;
+            if (update_result.modified_at_conflict) {
+              error_count++;
+              msg += `\nERROR modified at conflict for ${instance_question.id}`;
+            }
+          } catch (err) {
+            logger.error('error while regrading', { err });
+            error_count++;
+            msg = `ERROR bot grading for ${instance_question.id}`;
+          }
+          output = (output == null ? '' : `${output}\n`) + msg;
+          output_count++;
+          if (output_count >= 100) {
+            job.info(output);
+            output = null;
+            output_count = 0;
+          }
+        }
+
+        if (output != null) {
+          job.info(output);
+        }
+        if (error_count > 0) {
+          job.error('Number of errors: ' + error_count);
+          job.fail('Errors occurred while bot grading, see output for details');
+        }
+      });
+
+      // for debugging, run your docker container with "docker run -it --rm -p 3000:3000 -e NODEMON=true -v ~/git/PrairieLearn:/PrairieLearn --name mypl prairielearn/prairielearn"
+      // to check out your database, run "docker exec -it mypl psql postgres"
+
+      res.redirect(req.originalUrl);
```

Comment:
I believe a better strategy here would be to redirect to the job sequence page, so that the output can be followed.